runtime/client/client.go

```diff
@@ -61,9 +65,23 @@ func (o *Options) BindFlags(fs *pflag.FlagSet) {
 		"The maximum burst queries-per-second of requests sent to the Kubernetes API.")
 }
 
-// GetConfigOrDie wraps ctrl.GetConfigOrDie and sets the configured Options, returning the modified rest.Config.
+// GetConfigOrDie wraps ctrl.GetConfigOrDie and checks if the Kubernetes apiserver
+// has PriorityAndFairness flow control filter enabled. If true, it returns a rest.Config
+// with client side throttling disabled. Otherwise, it returns a modified rest.Config
+// configured with the provided Options.
 func GetConfigOrDie(opts Options) *rest.Config {
```

Comment:
While working on https://github.com/fluxcd/helm-controller/pull/480, I discovered this func does not have a non-panicing counterpart.

I think this would be useful to have in situations where you work with the option, but not the configuration (like the above PR).